File: app/src/main/java/mb00/android/codehub/logic/preferences/PreferenceManagerImpl.java
package mb00.android.codehub.logic.preferences;


import android.content.Context;
import android.content.SharedPreferences;

import mb00.android.codehub.data.PreferenceKeys;

public class PreferenceManagerImpl implements PreferenceManager {

    private SharedPreferences sharedPreferences;

    public PreferenceManagerImpl(final Context context) {
        sharedPreferences = context.getSharedPreferences(PreferenceKeys.PREFERENCES, Context.MODE_PRIVATE);
    }

    @Override
    public boolean isSignedIn() {
        return sharedPreferences.getBoolean(PreferenceKeys.SIGNED_IN, false);
    }

    @Override
    public void setSignedIn(boolean isSignedIn) {
        sharedPreferences.edit().putBoolean(PreferenceKeys.SIGNED_IN, isSignedIn).apply();
    }

    @Override
    public String getUsername() {
        return sharedPreferences.getString(PreferenceKeys.USER_NAME, "");
    }

    @Override
    public void setUsername(String username) {
        sharedPreferences.edit().putString(PreferenceKeys.USER_NAME, username).apply();
    }

    @Override
    public String getAuthHeader() {
        return sharedPreferences.getString(PreferenceKeys.AUTH_HEADER, "");
    }

    @Override
    public void setAuthHeader(String authHeader) {
        sharedPreferences.edit().putString(PreferenceKeys.AUTH_HEADER, authHeader).apply();
    }


}

File: app/src/main/java/mb00/android/codehub/logic/preferences/PreferenceManagerImpl.kt
package mb00.android.codehub.logic.preferences

import android.content.Context
import android.content.SharedPreferences

import mb00.android.codehub.data.PreferenceKeys


class PreferenceManagerImpl(context: Context) : PreferenceManager {

    private val sharedPreferences: SharedPreferences = context.getSharedPreferences(PreferenceKeys.PREFERENCES, Context.MODE_PRIVATE)

    override var isSignedIn: Boolean
        get() = sharedPreferences.getBoolean(PreferenceKeys.SIGNED_IN, false)
        set(isSignedIn) = sharedPreferences.edit().putBoolean(PreferenceKeys.SIGNED_IN, isSignedIn).apply()

    override var username: String
        get() = sharedPreferences.getString(PreferenceKeys.USER_NAME, "")
        set(username) = sharedPreferences.edit().putString(PreferenceKeys.USER_NAME, username).apply()

    override var authHeader: String
        get() = sharedPreferences.getString(PreferenceKeys.AUTH_HEADER, "")
        set(authHeader) = sharedPreferences.edit().putString(PreferenceKeys.AUTH_HEADER, authHeader).apply()

    override var accessToken: String
        get() = sharedPreferences.getString(PreferenceKeys.ACCESS_TOKEN, "")
        set(accessToken) = sharedPreferences.edit().putString(PreferenceKeys.ACCESS_TOKEN, accessToken).apply()

}
